src/peergos/server/tests/simulation/Simulator.java

```diff
@@ -535,13 +642,23 @@ private boolean verifySharingPermissions(String user, Path path) {
     private boolean verifyContents(String user, Path path) {
         FileSystem peergosFileSystem = fileSystems.getTestFileSystem(user);
         FileSystem nativeFileSystem = fileSystems.getReferenceFileSystem(user);
-
+        Stat stat = fileSystems.getReferenceFileSystem(user).stat(path);
+        boolean isDir = stat.fileProperties().isDirectory;
         try {
-            byte[] testData = peergosFileSystem.read(path);
-            byte[] refData = nativeFileSystem.read(path);
-            if (!Arrays.equals(testData, refData)) {
-                LOG.info("Path " + path + " has different contents between the file-systems");
-                return false;
+            if (isDir) {
+                Set<Path> refDirectoryListing = new TreeSet<>(nativeFileSystem.ls(path, true));
```

Comment:
I'm not sure this  is necessary -  by checking  each  _file_ in both filesystems the list of files in each directory is also implicitly checked.

What this will check is if there is some problem with the  result of the `ls` call in each directory - a view that should be the same as reading all the files we think are there. If we're going to check  `ls`  explicitly I think do  it separately - iterate through  all the directories and check the `ls` rather  than shoe-horning it in here.

Reply:
You can be granted read access to things in a directory and not have full read access to the parent dir, this means you wouldn't be able to see siblings of the files. Similarly you can be granted write access to a file, but not even read to its parent. This means you can modify that file, but not rename it, nor ls the parent dir. This is what unix does.

Reply:
Yes, and this should still be checked separately.

Reply:
What about an empty directory? Just checking files won't catch that.